test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/DataAnnotationsClientModelValidatorProviderTest.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    public class DataAnnotationsClientModelValidatorProviderTest
+    {
+        private readonly IModelMetadataProvider _metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+
+        [Fact]
+        public void GetValidators_DoesNotAddRequiredAttribute_IfAttributeIsSpecifiedExplicitly()
+        {
+            // Arrange
+            var provider = new DataAnnotationsClientModelValidatorProvider();
+            var metadata = _metadataProvider.GetMetadataForProperty(typeof(DummyRequiredAttributeHelperClass),
+                                                                    "WithAttribute");
```

Comment:
side note: this is test code but don't get into the habit of wrapping lines this way. (a) just hit `<Enter>` (don't force the next person to enter all the spaces) and (b) if you wrap one argument, wrap them all.